### ISSUE TITLE
Image with 'latest' tag corresponds to an older etcd version

### ISSUE THREAD

**Issue by nickbel7** [etcd-io/etcd#15382]
### What happened?

I tried to deploy an etcd cluster as a StatefulSet using the etcd image from the [quay.io/coreos/etcd](http://quay.io/coreos/etcd) repository, using the 'latest' tag, as proposed in the following documentation (https://etcd.io/docs/v2.3/docker_guide/).
The etcd version that I get in each pod is the v3.3.8 (neither 3.4 or 3.5) which is a older version of etcd.

Is this something that has been done on purpose because of the parallel maintenance of the 3.4 and 3.5 versions ?
If so, I would expect that the default etcd documentation does not prompt to use the latest tag (or state that this is not a good practice).

### What did you expect to happen?

I would expect that the newly created etcd pods would have the version tagged as 'latest' in the github releases page (https://github.com/etcd-io/etcd/releases). 

### How can we reproduce it (as minimally and precisely as possible)?

(Fast way) Docker
    - Get the 'latest' etcd image 
    `docker pull quay.io/coreos/etcd:latest`
    - Get the etcd image id from `docker images`
    - Try to create a container
    `docker run <image_id>`
    - Observe the etcd version in the logs
    (ex. `2023-03-01 11:31:18.023271 I | etcdmain: etcd Version: 3.3.8`)

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
etcd Version: 3.3.8
Git SHA: 33245c6b5
Go Version: go1.9.7
Go OS/Arch: linux/amd64

$ etcdctl version
etcdctl version: 3.3.8
API version: 2
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

### Relevant log output

```shell
$ kubectl exec -it etcd-0 -- etcd --version
etcd Version: 3.3.8
```

**Comment by serathius** [comment]
Looks like it's a repeat of https://github.com/etcd-io/etcd/issues/13606 but for quey repo.

**Comment by nickbel7** [comment]
> you're browsing the documentation from etcd 2.3
@tjungblu You are right, I was aware of that. However, I didn't find the same page in earlier versions in the documentation.
In v3.5 for example, I noticed the installation process regarding Kubernetes is incomplete (https://etcd.io/docs/v3.5/install/).
Is there some updated page for running etcd under Docker which I am missing ?

**Comment by ahrtr** [comment]
> > you're browsing the documentation from etcd 2.3
> 
> @tjungblu You are right, I was aware of that. However, I didn't find the same page in earlier versions of the documentation. In v3.5 for example, I noticed the installation process regarding Kubernetes is incomplete (https://etcd.io/docs/v3.5/install/). As I understand, someone would have to go to the github releases page in order to see that `gcr.io/etcd-development/etcd` is the primary container registry

There is a feature request to address this. https://github.com/etcd-io/etcd/issues/15185 
It looks like we can update the guide https://etcd.io/docs/v3.5/install/.

**Comment by jmhbnz** [comment]
Hey @ahrtr,  @serathius do we have permissions to be able to remove `latest` from quay.io?

There was a comment on https://github.com/etcd-io/etcd/issues/13606, users are still running into confusion in relation to `latest`.  My vote would also go to just removing it so we can close out this issue and hopefully prevent some confusion.

**Comment by serathius** [comment]
Sure

**Comment by jmhbnz** [comment]
`latest` tag has been deleted from `quay.io/coreos/etcd`, closing.
